suite/tests/api/drdecode_x86.c

```diff
@@ -155,12 +157,20 @@ test_noalloc(void)
      */
 }
 
-#define CHECK_CATEGORY(dcontext, instr, pc, category)                          \
-    ASSERT(instr_encode(dcontext, instr, pc) - pc < BUFFER_SIZE_ELEMENTS(pc)); \
-    instr_reset(dcontext, instr);                                              \
-    instr_set_operands_valid(instr, true);                                     \
-    ASSERT(decode(dcontext, pc, instr) != NULL);                               \
-    ASSERT(instr_get_category(instr) == category);                             \
+#define CHECK_CATEGORY(dcontext, instr, pc, categories, category_names)           \
+    ASSERT(instr_encode(dcontext, instr, pc) - pc < BUFFER_SIZE_ELEMENTS(pc));    \
```

Comment:
Putting key steps needed for subsequent code inside something like ASSERT which is sometimes debug-only and turns into a nop in release build can be dangerous as the subsequent code will then fail.  Here it looks like ASSERT is defined in release build too.  But imagine this code being copied/modeled in some other file where that's not the case.  So it may be worth making safer code by separating the instr_encode and decode calls from the asserts.  Or add a comment stating the assumption that ASSERT is always enabled.

Reply:
Done.